test/unit/spec/localparticipant.js

```diff
@@ -151,6 +155,123 @@ describe('LocalParticipant', () => {
     });
   });
 
+  describe('#setBandwidthProfile', () => {
+    const subscriptionModes = Object.values(subscriptionMode);
+    const trackPriorities = [PRIORITY_HIGH, PRIORITY_LOW, PRIORITY_STANDARD];
+
+    [
+      [null, 'that is null', TypeError, 'object'],
+      ['foo', 'that is not an object', TypeError, 'object'],
+      [['bar'], 'that is an Array', TypeError, 'object'],
+      [{ video: null }, 'whose .video is null', TypeError, 'object'],
```

Comment:
please add `{ video: undefined }` as a test case to make sure that gets handled as expected.

Reply:
@makarandp0 `JSON.stringify()` will remove properties explicitly set to `undefined`. So I think we are good here.

Reply:
@manjeshbhargav, I am sure it works now, but adding a test would ensure that if we manage to break the behavior it will continues to work.

Reply:
@makarandp0 I've already added unit tests for the cases you mentioned in the last commit.